crates/metadata-store/src/omnipaxos/store.rs

```diff
@@ -0,0 +1,1232 @@
+            {
+                for (idx, decided_entry) in decided_entries.into_iter().enumerate() {
+                    match decided_entry {
+                        LogEntry::Decided(request) => {
+                            self.kv_storage.handle_request(request);
+                        }
+                        LogEntry::Undecided(_) => {
+                            panic!("Unexpected undecided entry")
+                        }
+                        LogEntry::Trimmed(_) => {
+                            unimplemented!("We don't support trimming yet")
+                        }
+                        LogEntry::Snapshotted(_) => {
+                            unimplemented!("We don't support snapshots yet")
+                        }
+                        LogEntry::StopSign(ss, decided) => {
+                            assert!(decided, "we are handling only decided entries");
+                            assert_eq!(
+                                (idx + 1) + self.last_applied_index,
+                                last_decided_index,
+                                "StopSigns must be the last decided entries"
+                            );
+                            return self.handle_stop_sign(ss);
+                        }
+                    }
+                }
+            }
+
+            self.last_applied_index = last_decided_index;
+        }
+
+        Ok(DecidedEntriesResult::Continue)
+    }
+
+    fn handle_stop_sign(
+        &mut self,
+        stop_sign: StopSign,
+    ) -> Result<DecidedEntriesResult, DecidedEntriesError> {
+        trace!("Handling stop sign");
+
+        // remove the stop sign from the entries that are applied to the kv_storage because it is
+        // virtual entry that is not part of the log
+        let last_decided_index = self
+            .omni_paxos
+            .as_ref()
+            .expect("to be present")
+            .get_decided_idx()
+            - 1;
+        self.last_applied_index = last_decided_index;
+
+        let metadata = Self::deserialize_omni_paxos_config_metadata(stop_sign.metadata.as_ref());
+
+        if Self::is_member(self.own_member_id, &metadata.members) {
+            debug!(configuration_id = %stop_sign.next_config.configuration_id, "Continue as part of new configuration: {:?}", stop_sign.next_config.nodes);
+
+            let mut rocksdb_storage = self.omni_paxos.take().expect("be present").into_inner();
+
+            let new_cluster_config = stop_sign.next_config.clone();
+            let omni_paxos_configuration = OmniPaxosConfiguration {
+                own_member_id: self.own_member_id,
+                cluster_config: stop_sign.next_config,
+                members: metadata.members,
+            };
+
+            Self::reset_storage_for_new_configuration(
+                &mut rocksdb_storage,
+                &omni_paxos_configuration,
+                last_decided_index,
+            )?;
```

Comment:
Stop signs are not part of the actual log. Therefore, we need to manually reset the decided index and clear some fields in the `RocksDbStorage` in order to reuse the same storage for the next configuration.